spec/addressable/uri_spec.rb

```diff
@@ -6799,3 +6799,23 @@ def to_str
     expect(res).to be nil
   end
 end
+
+describe Addressable::URI, "subdomain" do
+  it "should return the first subdomain" do
+    expect(Addressable::URI.parse("http://www.test.example.org").subdomain).to eq("www")
+  end
+
+  it "should return nil if no subdomain" do
+    expect(Addressable::URI.parse("http://example.co.uk").subdomain).to be_nil
+  end
+end
+
+describe Addressable::URI, "subdomains" do
+  it "should return the subdomains" do
+    expect(Addressable::URI.parse("http://www.test.example.org").subdomains).to eq(["www", "test"])
+  end
+
+  it "should return an empty array if no subdomain" do
+    expect(Addressable::URI.parse("http://example.co.uk").subdomains).to eq([])
+  end
+end
```

Comment:
Please keep the [terminating newline](https://stackoverflow.com/a/729795/525616)

<img width="227" alt="Screenshot 2023-06-03 at 22 18 31" src="https://github.com/sporkmonger/addressable/assets/42626/03e1c310-404e-4e06-b9e9-14ef4200d0e5">